core/io/http_client.cpp

```diff
@@ -515,34 +527,53 @@ PoolByteArray HTTPClient::read_response_body_chunk() {
 
 	} else {
 
-		int to_read = MIN(body_left, read_chunk_size);
+		int to_read = body_size != -1 ? MIN(body_left, read_chunk_size) : read_chunk_size;
 		PoolByteArray ret;
 		ret.resize(to_read);
 		int _offset = 0;
-		while (to_read > 0) {
+		while (body_size == -1 || to_read > 0) {
 			int rec = 0;
 			{
 				PoolByteArray::Write w = ret.write();
 				err = _get_http_data(w.ptr() + _offset, to_read, rec);
 			}
 			if (rec > 0) {
-				body_left -= rec;
-				to_read -= rec;
 				_offset += rec;
+				if (body_size != -1) {
+					body_left -= rec;
+					to_read -= rec;
+				} else {
+					if (rec < to_read) {
+						ret.resize(_offset);
```

Comment:
This gets really deeply nested, since the outmost else always breaks can you invert the condition of the outer `if`?
i.e.:
```
if (rec <= 0) {
	if (to_read > 0) // Ended up reading less
		ret.resize(_offset);
	break; // Nothing left to read
}
// code for rec > 0
```

Reply:
Done!